src/controllers/user.js

```diff
@@ -0,0 +1,59 @@
+import Model from '../models';
+
+const { Users } = Model;
+
+/**
+* Generic update profile
+
+* @function
+*
+* @param {request} request - express request object
+*
+* @return {Object} - callback that execute the controller
+*
+* @description allows users to update their profile
+*/
+const updateProfile = async (request) => {
+  const {
+    params: { id },
+    body: payload
+  } = request;
+
+  const [numberOfAffectedRows, affectedRows] = await Users.update(
+    { ...payload },
+    { where: { id }, returning: true, raw: true }
```

Comment:
What would have if the user doesn't exist? Or if another user wants to update a user's profile?

Reply:
I have added the logic for the scenario where a user doesn't exist. I will need a function from the middleware branch for the implementation of when a user updates a profile that is not theirs.